data/pigui/modules/map-sector-view.lua

```diff
@@ -0,0 +1,229 @@
+								if ui.collapsingHeader(label .. ': ' .. starsystem.name .. '  ' .. jumpData .. " (" .. math.floor(systempath.sectorX) .. ", " .. math.floor(systempath.sectorY) .. ", " .. math.floor(systempath.sectorZ) .. ")", { "DefaultOpen" }) then
+									local stars = starsystem:GetStars()
+									for _,star in pairs(stars) do
+										if ui.selectable(star.name, star.path == systempath, {}) then
+											clicked = star.path
+										end
+									end
+									if clicked then
+										Engine.SetSectorMapSelected(clicked)
+										Engine.SectorMapGotoSystemPath(clicked)
+									end
+									local numstars = starsystem.numberOfStars
+									local numstarstext = ""
+									if numstars == 4 then
+										numstarstext = lc.QUADRUPLE_SYSTEM
+									elseif numstars == 3 then
+										numstarstext = lc.TRIPLE_SYSTEM
+									elseif numstars == 2 then
+										numstarstext = lc.BINARY_SYSTEM
+									else
+										numstarstext = starsystem.rootSystemBody.astroDescription
+									end
+									ui.text(numstarstext)
+									ui.text(starsystem.shortDescription)
+									if othersystempath and not othersystempath:IsSameSystem(systempath) then
+										local otherstarsystem = othersystempath:GetStarSystem()
+										local jumpStatus, distance, fuelRequired, duration = player:GetHyperspaceDetails(systempath, othersystempath)
+										ui.text('Relative to ' .. otherstarsystem.name .. ': ' .. jumpStatus .. "  " .. string.format("%.2f", distance) .. lc.UNIT_LY .. "  " .. fuelRequired .. lc.UNIT_TONNES .. "  " .. ui.Format.Duration(duration, 2))
+									end
+								end
+		end)
+	end
+end
+
+local search_text = ""
+local draw_vertical_lines = false
+local draw_out_range_labels = false
+local draw_uninhabited_labels = true
+local automatic_system_selection = true
+local lock_hyperspace_target = false
+
+local initialized
+local function showSettings()
+	if ui.collapsingHeader("Settings", { "DefaultOpen" }) then
+		local changed
+		changed, draw_vertical_lines = ui.checkbox(lc.DRAW_VERTICAL_LINES, draw_vertical_lines)
+		if changed then
+			Engine.SetSectorMapDrawVerticalLines(draw_vertical_lines)
+		end
+		changed, draw_out_range_labels = ui.checkbox(lc.DRAW_OUT_RANGE_LABELS, draw_out_range_labels)
+		if changed then
+			Engine.SetSectorMapDrawOutRangeLabels(draw_out_range_labels)
+		end
+		changed, draw_uninhabited_labels = ui.checkbox(lc.DRAW_UNINHABITED_LABELS, draw_uninhabited_labels)
+		if changed then
+			Engine.SetSectorMapDrawUninhabitedLabels(draw_uninhabited_labels)
+		end
+		changed, automatic_system_selection = ui.checkbox(lc.AUTOMATIC_SYSTEM_SELECTION, automatic_system_selection)
+		if changed then
+			Engine.SetSectorMapAutomaticSystemSelection(automatic_system_selection)
+		end
+	end
+end
+
+local hyperspaceDetailsCache = {}
+
+local function getHyperspaceDetails(path)
+	local p = path.sectorX .. "/" .. path.sectorY .. "/" .. path.sectorZ .. "/" .. path.systemIndex
+	local it = hyperspaceDetailsCache[p]
```

Comment:
The cache should be invalidated in case of equipment change (and ship change, of course).